test/e2e/node_feature_discovery.go

```diff
@@ -190,65 +247,182 @@ func nfdMasterPod(ns string, image string, onMasterNode bool) *v1.Pod {
 	return p
 }
 
-func nfdWorkerPod(ns string, image string) *v1.Pod {
-	return &v1.Pod{
+func nfdWorkerPod(image string, extraArgs []string) *v1.Pod {
+	p := &v1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: "nfd-worker-" + string(uuid.NewUUID()),
 		},
-		Spec: v1.PodSpec{
-			// NOTE: We omit Volumes/VolumeMounts, at the moment as we only test the fake source
-			Containers: []v1.Container{
-				{
-					Name:            "node-feature-discovery",
-					Image:           image,
-					ImagePullPolicy: v1.PullAlways,
-					Command:         []string{"nfd-worker"},
-					Args:            []string{"--oneshot", "--sources=fake", "--server=nfd-master-e2e:8080"},
-					Env: []v1.EnvVar{
-						{
-							Name: "NODE_NAME",
-							ValueFrom: &v1.EnvVarSource{
-								FieldRef: &v1.ObjectFieldSelector{
-									FieldPath: "spec.nodeName",
-								},
+		Spec: nfdWorkerPodSpec(image, extraArgs),
+	}
+
+	p.Spec.RestartPolicy = v1.RestartPolicyNever
```

Comment:
Why do we set this outside of the `nfdWorkerPodSpec` function?

Reply:
That's because we use the same podspec (returned by `nfdWorkerPodSpec()` for the daemonset and daemonset may not have `RestartPolicyNever`